src/eShop.ServiceDefaults/ConfigureSwaggerOptions.cs

```diff
@@ -0,0 +1,178 @@
+using Microsoft.Extensions.Options;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace eShop.ServiceDefaults;
+
+internal sealed class ConfigureSwaggerOptions : IConfigureOptions<SwaggerGenOptions>
+{
+    private readonly IApiVersionDescriptionProvider _provider;
+    private readonly IConfiguration _configuration;
+
+    public ConfigureSwaggerOptions(IApiVersionDescriptionProvider provider, IConfiguration configuration)
+    {
+        _provider = provider;
+        _configuration = configuration;
+    }
+
+    public void Configure(SwaggerGenOptions options)
+    {
+        foreach (var description in _provider.ApiVersionDescriptions)
+        {
+            options.SwaggerDoc(description.GroupName, CreateInfoForApiVersion(description));
+        }
+
+        ConfigureAuthorization(options);
+    }
+
+    private OpenApiInfo CreateInfoForApiVersion(ApiVersionDescription description)
+    {
+        /// {
+        ///   "OpenApi": {
+        ///     "Document": {
+        ///         "Title": ..
+        ///         "Version": ..
+        ///         "Description": ..
+        ///     }
+        ///   }
+        /// }
+        var openApi = _configuration.GetSection("OpenApi");
+        var document = openApi.GetRequiredSection("Document");
+        var info = new OpenApiInfo()
+        {
+            Title = document.GetRequiredValue("Title"),
+            Version = description.ApiVersion.ToString(),
+            Description = BuildDescription(description, document.GetRequiredValue("Description")),
+        };
+
+        return info;
+    }
+
+    private static string BuildDescription(ApiVersionDescription api, string description)
+    {
+        var text = new StringBuilder(description);
+
+        if (api.IsDeprecated)
+        {
+            if (text.Length > 0)
+            {
+                if (text[^1] == '.')
+                {
+                    text.Append('.');
+                }
+
+                text.Append(' ');
+            }
+
+            text.Append("This API version has been deprecated.");
+        }
+
+        if (api.SunsetPolicy is { } policy)
+        {
+            if (policy.Date is { } when)
+            {
+                if (text.Length > 0)
+                {
+                    text.Append(' ');
+                }
+
+                text.Append("The API will be sunset on ")
+                    .Append(when.Date.ToShortDateString())
+                    .Append('.');
+            }
+
+            if (policy.HasLinks)
+            {
+                text.AppendLine();
+
+                foreach (var link in policy.Links.Where(l => l.Type == "text/html"))
+                {
+                    text.AppendLine();
+
+                    if (link.Title.HasValue)
+                    {
+                        text.Append(link.Title.Value).Append(": ");
+                    }
+
+                    text.Append(link.LinkTarget.OriginalString);
```

Comment:
Small quality of life improvement to make the link clickable!
```suggestion
                    text.Append($"<{link.LinkTarget.OriginalString}>");
```

Reply:
Nice suggestion. I don't think adding `<` and `>` alone will do the trick; however, I've refactored the implementation to render an unordered list of links, if any. If any links for HTML pages are defined, it will render like so:

```html
<h4>Links</h4>
<ul>
 <li><a href="policy.html">Versioning Policy</a></li>
</ul>
```

The [API Versioning examples](https://github.com/dotnet/aspnet-api-versioning/tree/main/examples) for OpenAPI have been updated to reflect this change as well.